src/borg/testsuite/archiver.py

```diff
@@ -3186,6 +3186,11 @@ def test_with_lock(self):
         cmd = 'python3', '-c', 'import os, sys; sys.exit(42 if os.path.exists("%s") else 23)' % lock_path
         self.cmd('with-lock', self.repository_location, *cmd, fork=True, exit_code=42)
 
+    def test_with_lock_non_existent_command(self):
+        self.cmd('init', '--encryption=repokey', self.repository_location)
+        cmd = 'non_existant_command'
+        self.cmd('with-lock', self.repository_location, *cmd, fork=True, exit_code=EXIT_ERROR)
```

Comment:
Guess that *cmd does not quite work as you maybe thought.

Try it in an interactive python interpreter.

Reply:
Also not sure what you mean:
```bash
(venv) kmille@linbox:borg pytest src/borg/testsuite/archiver.py -k with_lock_non -v -s   
...                                                                                                                                                                                          
src/borg/testsuite/archiver.py::ArchiverTestCase::test_with_lock_non_existent_command PASSED
src/borg/testsuite/archiver.py::ArchiverTestCaseBinary::test_with_lock_non_existent_command SKIPPED (no borg.exe available)
src/borg/testsuite/archiver.py::RemoteArchiverTestCase::test_with_lock_non_existent_command PASSED

================================================================================================ 2 passed, 1 skipped, 677 deselected in 2.49s =================================================================================================
(venv) kmille@linbox:borg 
```

Reply:
```
tw@mba2020 ~ % python3
Python 3.11.6 (main, Oct  2 2023, 13:45:54) [Clang 15.0.0 (clang-1500.0.40.1)] on darwin

>>> cmd = 'non_existant_command'

>>> print(*cmd)
n o n _ e x i s t a n t _ c o m m a n d
```

Yes, if something is clear / agreed / resolved in the code, you click that button.

Reply:
I thought about that, but the test run without an issue :)
If you like I can add a comma: `cmd = 'non_existent_command', `

Reply:
A comma would work, but guess a list would be clearest:
```
cmd = ['non_existent_command']
```